Tasks/KubernetesV1/task.json

```diff
@@ -0,0 +1,404 @@
+            "visibleRule": "useConfigMapFile = true",
+            "groupName": "configMaps"	
+        },
+        {
+            "name": "configMapArguments",
+            "type": "multiLine",
+            "properties": {
+                "resizable": "true",
+                "rows": "2",
+                "editorExtension": "ms.vss-services-azure.kubernetes-parameters-grid"
+            },
+            "label": "Arguments",
+            "helpMarkDown": "Specify keys and literal values to insert in configMap.For example, --from-literal=key1=value1 --from-literal=key2=\"top secret\".",
+            "visibleRule": "useConfigMapFile = false",
+            "groupName": "configMaps"
+        },
+        {
+            "name": "versionOrLocation",
+            "type": "radio",
+            "label": "Kubectl",
+            "defaultValue": "version",
+            "required": false,
+            "options": {
+                "version": "Version",
+                "location": "Specify location"
+            },
+            "groupName": "advanced"
+        },
+        {
+            "name": "versionSpec",
+            "type": "string",
+            "label": "Version spec",
+            "defaultValue": "1.7.0",
+            "helpMarkDown": "Version Spec of version to get. Examples: 1.7.0, 1.x.0, 4.x.0, 6.10.0, >=6.10.0",
+            "groupName": "advanced",
+            "visibleRule": "versionOrLocation = version"
+        },
+        {
+            "name": "checkLatest",
+            "type": "boolean",
+            "label": "Check for latest version",
+            "defaultValue": "false",
+            "helpMarkDown": "Always checks online for the latest available version (stable.txt) that satisfies the version spec. This is typically false unless you have a specific scenario to always get latest. This will cause it to incur download costs when potentially not necessary, especially with the hosted build pool.",
+            "required": false,
+			"groupName": "advanced",
+			"visibleRule": "versionOrLocation = version"
+        },
+        {
+            "name": "specifyLocation",
+            "type": "filePath",
+            "label": "Path to Kubectl",
+            "defaultValue": "",
+            "helpMarkDown": "Full path to the kubectl.exe",
+            "required": true,
+            "groupName": "advanced",
+            "visibleRule": "versionOrLocation = location"
+        },
+        {
+            "name": "cwd",
+            "aliases": ["workingDirectory"],
+            "type": "filePath",
+            "label": "Working directory",
+            "defaultValue": "$(System.DefaultWorkingDirectory)",
+            "helpMarkDown": "Working directory for the Kubectl command.",
+            "groupName": "advanced"
+        }       
+    ],
+    "dataSourceBindings": [
+        {
+            "target": "azureContainerRegistry",
+            "endpointId": "$(azureSubscriptionEndpoint2)",
```

Comment:
If user has selected the AKS connection type and user selects docker registry in secrets, do we really need to ask for Azure container registry and azure service endpoint.